homeassistant/helpers/translation.py

```diff
@@ -101,10 +102,12 @@ async def _async_get_component_strings(
     integrations: dict[str, Integration],
 ) -> dict[str, dict[str, Any]]:
     """Load translations."""
-    translations_by_language: dict[str, dict[str, Any]] = {}
+    translations_by_language: defaultdict[str, dict[str, Any]] = defaultdict(dict)
     # Determine paths of missing components/platforms
     files_to_load_by_language: dict[str, dict[str, pathlib.Path]] = {}
-    loaded_translations_by_language: dict[str, dict[str, Any]] = {}
+    loaded_translations_by_language: defaultdict[str, defaultdict[str, Any]] = (
```

Comment:
This is a variable that is only populated by the output of another function, and the only purpose on this method is to query for certain values in order to return them into the `translations_by_language` variable. Therefore it would be better to not target to any mutation on this variable as we don't plan to make any insertion. This would mean:

1. Mark this as a standalone `dict`.
2. Instead of accessing by with `[key]` (which would mutate the underlying object if the key is missing), use `.get(key, {})` so you don't mutate the returned object.